pkg/daemon/ceph/osd/volume.go

```diff
@@ -877,21 +877,43 @@ func (a *OsdAgent) WipeDevicesFromOtherClusters(context *clusterd.Context) error
 	for _, existingOSD := range existingOSDs {
 		// Wipe the devices that will be used for preparing OSD but already have OSD metadata from another ceph cluster
 		if existingOSD.CephFsid != a.clusterInfo.FSID {
-			device := desiredDeviceFromOtherCluster(context.Devices, existingOSD.Device)
```

Comment:
nit: this block is getting long, it might be nice to factor out to a helper method for wiping a single device